klab.hub/src/main/java/org/integratedmodelling/klab/hub/api/Agreement.java

```diff
@@ -167,8 +168,25 @@ public String getOwnAgreement() {
     public void setOwnAgreement(String ownAgreement) {
         this.ownAgreement = ownAgreement;
     }
-    
-    
-        
+
+    public boolean isRevoked() {
+        return revokedDate != null;
+    }
+
+    public boolean hasExpirationDate() {
+        // Agreements with null expiration dates are the ones with no expiration date
+        return expiredDate != null;
+    }
+
+    public boolean isExpired() {
+        if(!hasExpirationDate()) {
+            return false;
+        }
+        return !expiredDate.toInstant().isAfter(Instant.now());
```

Comment:
I can see problem with the expiration date.
What do you think about `revokedDate` and `validDate`? If they have similar issues, maybe they could be changed too.
Also, a little bit out of topic, but it would be nice to have a way of representing a non-expiring `expirationDate` or a non revoked `revokedDate` without using a `null` value. We should think about that...